src/SonarScanner.MSBuild.PreProcessor/JreCaching/JreCache.cs

```diff
@@ -134,6 +135,19 @@ private async Task<Exception> DownloadJre(string jreDownloadPath, string downloa
         }
     }
 
+    private void TryDeleteFile(string tempFile)
+    {
+        try
+        {
+            logger.LogDebug(Resources.MSG_DeletingFile, tempFile);
+            fileWrapper.Delete(tempFile);
+        }
+        catch (Exception ex)
+        {
+            logger.LogDebug(Resources.MSG_DeletingFileFailure, tempFile, ex.Message);
+        }
+    }
+
     private static void EnsureClosed(Stream fileStream)
```

Comment:
The [documentation](https://learn.microsoft.com/en-us/dotnet/api/system.io.stream.close?view=net-8.0#:~:text=.%20Instead%20of%20calling%20this%20method%2C%20ensure%20that%20the%20stream%20is%20properly%20disposed) recommends:
>  Instead of calling this method, ensure that the stream is properly disposed.

I think it would be better to just call stream.Dispose(). This would also remove the need of using a try block; the Dispose shouldn't throw.

Reply:
I'm not sure about the non-throwing behaviour of FileStream.Close / Dispose on different platforms and neither are the runtime experts: https://github.com/dotnet/runtime/issues/17418
We need to close the stream because we can't delete the file unless it is closed. The important part here is: 

> Closes the current stream and releases any resources (such as sockets and file handles) associated with the current stream.

[Source](https://learn.microsoft.com/en-us/dotnet/api/system.io.filestream.close?view=netframework-1.1)

The file handle needs to be closed. If you remove the file's closing, the UT `Download_DownloadFileNew_Failure_WithTestFiles` will fail.

Reply:
.. and Dispose throws in at least some versions of the runtime:
https://github.com/dotnet/runtime/pull/38742/files#diff-0592eec0311ffa397e0bba57192f717710475f94a25440f137c513018a80cdf5R73-R76